src/Microsoft.Graph.Core/Tasks/LargeFileUploadTask.cs

```diff
@@ -33,24 +35,48 @@ public class LargeFileUploadTask<T>
         /// <param name="maxSliceSize">Max size of each slice to be uploaded. Multiple of 320 KiB (320 * 1024) is required.</param>
         /// <param name="baseClient"><see cref="BaseClient"/> to use for making upload requests. The client should not set Auth headers as upload urls do not need them.
         /// If less than 0, default value of 5 MiB is used. .</param>
-        public LargeFileUploadTask(IUploadSession uploadSession, Stream uploadStream,  int maxSliceSize = -1, BaseClient baseClient = null)
+        public LargeFileUploadTask(IParsable uploadSession, Stream uploadStream,  int maxSliceSize = -1, BaseClient baseClient = null)
```

Comment:
push the IParsable interface on IUploadSession, revert this change in the ctor, this way you don't have to juggle down below to get the actual session object.

Reply:
I agree that would be better. I will need to run some tests first then will action and resolve.

Reply:
Resolved via https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/pull/350/commits/a649908819c3c2e1f93e878bfbeb2236886b1b85